test/index.mjs

```diff
@@ -40,23 +44,101 @@ test ('assertions', () => {
+  fl.value (eq ({id: 1, disposed: 1})) (runHook (mockHook) (fl.resolve));
 
-  effect (runHook (mockHook2) (compose (resolve) (eq ({id: 2, disposed: false}))));
-  value (eq ({id: 2, disposed: false})) (runHook (mockHook2) (resolve));
+  effect (runHook (mockHook2) (compose (fl.resolve) (eq ({id: 2, disposed: 0}))));
+  fl.value (eq ({id: 2, disposed: 0})) (runHook (mockHook2) (fl.resolve));
 
   eq (sequential (mockParallelHook)) (mockHook);
 
-  value (eq ([{id: 1, disposed: true}, {id: 2, disposed: false}]))
-        (runHook (hookAll ([mockHook, mockHook2])) (resolve));
+  fl.value (eq ([{id: 1, disposed: 1}, {id: 2, disposed: 0}]))
+           (runHook (hookAll ([mockHook, mockHook2])) (fl.resolve));
 });
+
+const mockHooks = [ hook (mockAcquire (1)) (mockDispose)
+                  , hook (delay (mockAcquire (2))) (mockDispose)
+                  , hook (mockAcquire (3)) (mockDisposeAsync)
+                  , hook (delay (mockAcquire (4))) (mockDisposeAsync)
+                  , hook (mockAcquire (5)) (mockDispose)
+                  , hook (delay (mockAcquire (6))) (mockDispose)
+                  , hook (mockAcquire (7)) (mockDisposeAsync)
+                  , hook (delay (mockAcquire (8))) (mockDisposeAsync) ];
+
+test ('async assertions', () => Promise.all ([
+  equivalence (runHook (hook (delay (mockAcquire (1))) (mockDisposeAsync)) (fl.resolve))
+              (fl.resolve ({id: 1, disposed: 1})),
+
+  equivalence (runHook (hookAll (mockHooks)) (fl.resolve))
+              (fl.resolve (countTo (8) .map (deadResource))),
+
+  fl.promise (runHook (hookAll (mockHooks))
+                      (compose (fl.resolve) (eq (countTo (8) .map (liveResource))))),
+]));
+
+const choice = [true, false];
+choice.forEach(lr => choice.forEach (ld => choice.forEach (rr => choice.forEach (rd => choice.forEach (ldd => choice.forEach (rdd => choice.forEach (cr => choice.forEach (cd => {
+  const name = `left acquire ${
+    lr ? 'reject' : 'resolve'
+  } ${
+    ld ? 'delayed' : 'instantly'
+  }; right acquire ${
+    rr ? 'reject' : 'resolve'
+  } ${
+    rd ? 'delayed' : 'instantly'
+  }; left disposal resolve ${
+    ldd ? 'delayed' : 'instantly'
+  }; right disposal resolve ${
+    rdd ? 'delayed' : 'instantly'
+  }; consumption ${
+    cr ? 'reject' : 'resolve'
+  } ${
+    cd ? 'delayed' : 'instantly'
+  }`;
+
+  const leftAcquire = (ld ? delay : id) (lr ? fl.reject ('meh') : fl.resolve (41));
+  const rightAcquire = (rd ? delay : id) (rr ? fl.reject ('meh') : fl.resolve (inc));
+  const leftDisposal = compose (ldd ? delay : id) (fl.resolve);
+  const rightDisposal = compose (rdd ? delay : id) (fl.resolve);
+  const consume = compose (cd ? delay : id) (cr ? _ => fl.reject ('meh') : fl.resolve);
+
+  const expected = lr || rr || cr ? fl.reject ('meh') : fl.resolve (42);
+
+  const cancellable = (
+    (lr && !ld) ? (false) :
+    (rr && !rd) ? (!ld && ldd) :
+    (ld || rd || cd || ldd || rdd)
+  );
```

Comment:
A running pair of ParallelHooks are cancellable if:

- The left hook doesn't immediately fail on resource acquisition; and
- the right hook doesn't immediately fail on resource acquisition, with the exception that the left hook had already acquired its resource, and its disposal takes time; and
- at least one of the Futures involved are async.